src/awkward/operations/ak_from_raggedtensor.py

```diff
@@ -28,20 +28,43 @@ def from_raggedtensor(array):
 
 
 def _impl(array):
+    try:
+        import tensorflow as tf
+    except ImportError as err:
+        raise ImportError(
+            """to use ak.from_raggedtensor, you must install the 'tensorflow' package with:
+
+        pip install tensorflow
+or
+        conda install tensorflow"""
+        ) from err
+
     try:
         # get the flat values
-        content = array.flat_values.numpy()
+        content = array.flat_values
     except AttributeError as err:
         raise TypeError(
             """only RaggedTensor can be converted to awkward array"""
         ) from err
-    # convert them to ak.contents right away
+
+    # handle gpu and cpu instances separately
+    device = content.device
+
+    # since TensorFlow currently does not support
+    # int32 variables being placed on the GPU, use CPU for them instead
+    if content.dtype == tf.int32:
+        device = "cpu"
```

Comment:
But you're coming _from_ TensorFlow, right? If TensorFlow doesn't support integers on the GPU, then the `content` you just got is either already `device="cpu"` or not integers.

Also, you don't want to pick out just one type like this: what about int64? int16? int8? Unsigned integers? The code should apply some general policy for any possible input. (But if what you're saying about TensorFlow is correct, then the above isn't a "possible input.")

```suggestion
```

Reply:
Also, it looks like the TensorFlow device name for arrays in main memory is `"CPU:0"`, but that doesn't matter if you'll be deleting this part.

Reply:
after more research I finally found the exact thing I needed: https://github.com/tensorflow/tensorflow/issues/41307

so, just using `.backing_device` instead of `.device` will work!